components/Modal.jsx

```diff
@@ -0,0 +1,34 @@
+import { useRef, useEffect } from "react";
+import styles from "@/styles/components/modal.module.scss";
+
+export default function Modal({ content, isModalOpen, setIsModalOpen }) {
+	const modalRef = useRef(null);
```

Comment:
Suggest calling it `ref` instead of `modalRef` as it's a ref to the parent element of the component and the only ref.
On the other side, one can make the argument that it's more clear to define in-name to which the ref belongs.
You choice.